### ISSUE TITLE
Performance issue for negated queries

### ISSUE THREAD

**Issue by denzelem** [makandra/minidusen#19]
Very long lists in PostgreSQL perform terribly in `NOT IN`.  Two suggestions are to use [an anti-join or a subquery](https://stackoverflow.com/a/17038097).

Example for submitting a filter query "!0" on a customer table with approx. ~1,000,000 rows and a runtime > 30 minutes.

```
SELECT "customers".* FROM "customers" WHERE "customers"."id" NOT IN (
  SELECT "customers"."id"
  FROM "customers"
  WHERE "customers"."first_name" ILIKE '%0%'
     OR "customers"."last_name" ILIKE '%0%'
     OR "customers"."internal_note" ILIKE '%0%'
) LIMIT 10;
```

We saw this performance issue in an application where the negate operator was submitted by accident (the search field did not display, that we support this syntax). And blocked a few worker for too long time.

Wdyt?

**Comment by triskweline** [comment]
How would that query look like using the other approaches you're suggesting?

**Comment by denzelem** [comment]
I think it could look like this. But I was not able to test this for correctness yet.

```
SELECT "customers".* FROM "customers"
LEFT OUTER JOIN "customers" "excluded_customers" ON (
"customers"."id" = "excluded_customers"."id"
  AND (
    "excluded_customers"."first_name" ILIKE '%0%'
    OR "excluded_customers"."last_name" ILIKE '%0%'
    OR "excluded_customers"."internal_note" ILIKE '%0%'
  )
)
WHERE "excluded_customers"."id" IS NULL;
```

**Comment by kratob** [comment]
@denzelem 
I went for the ANTI-JOIN solution. Fix is released as 0.11.2. Note that I did not do any performance tests, so I'd be curious if this does fix your issue.

**Comment by denzelem** [comment]
@kratob The query now runs within `529.630 ms`, before it was > 30 Minutes.  Thanks for fixing this issue!
